pubsub/google/cloud/pubsub_v1/publisher/client.py

```diff
@@ -0,0 +1,164 @@
+    get sensible defaults.
+
+    Args:
+        batch_settings (~.pubsub_v1.types.BatchSettings): The settings
+            for batch publishing.
+        batch_class (class): A class that describes how to handle
+            batches. You may subclass the
+            :class:`.pubsub_v1.publisher.batch.base.BaseBatch` class in
+            order to define your own batcher. This is primarily provided to
+            allow use of different concurrency models; the default
+            is based on :class:`multiprocessing.Process`.
+        kwargs (dict): Any additional arguments provided are sent as keyword
+            arguments to the underlying
+            :class:`~.gapic.pubsub.v1.publisher_client.PublisherClient`.
+            Generally, you should not need to set additional keyword arguments.
+    """
+    def __init__(self, batch_settings=(), batch_class=mp.Batch, **kwargs):
+        # Add the metrics headers, and instantiate the underlying GAPIC
+        # client.
+        kwargs['lib_name'] = 'gccl'
+        kwargs['lib_version'] = __VERSION__
+        self.api = publisher_client.PublisherClient(**kwargs)
+        self.batch_settings = types.BatchSettings(*batch_settings)
+
+        # The batches on the publisher client are responsible for holding
+        # messages. One batch exists for each topic.
+        self._batch_class = batch_class
+        self._batches = {}
+
+    @property
+    def concurrency(self):
+        """Return the concurrency strategy instance.
+
+        Returns:
+            ~.pubsub_v1.concurrency.base.PublishStrategy: The class responsible
+                for handling publishing concurrency.
+        """
+        return self._concurrency
+
+    def batch(self, topic, create=True, autocommit=True):
+        """Return the current batch.
+
+        This will create a new batch only if no batch currently exists.
+
+        Args:
+            topic (str): A string representing the topic.
+            create (bool): Whether to create a new batch if no batch is
+                found. Defaults to True.
+            autocommit (bool): Whether to autocommit this batch.
+                This is primarily useful for debugging.
+
+        Returns:
+            :class:~`pubsub_v1.batch.Batch` The batch object.
+        """
+        # If there is no matching batch yet, then potentially create one
+        # and place it on the batches dictionary.
+        accepting = base.BaseBatch.Status.ACCEPTING_MESSAGES
+        if self._batches.get(topic, base.FAKE).status != accepting:
+            if not create:
+                return None
+            self._batches[topic] = self._batch_class(
+                autocommit=autocommit,
+                client=self,
+                settings=self.batch_settings,
+                topic=topic,
+            )
+
+        # Simply return the appropriate batch.
+        return self._batches[topic]
+
+    def publish(self, topic, data, **attrs):
```

Comment:
Correct. We decided that you need to create multiple topic objects if you want to do this.